### PR TITLE
Dev/8.0

### PR DESCRIPTION
**Pull request by clpetersonucf**
- [#1310](https://github.com/ucfopen/Materia/pull/1310) - Docker readme updates
- [#1308](https://github.com/ucfopen/Materia/pull/1308) - Bump decompress from 4.2.0 to 4.2.1
- [#1304](https://github.com/ucfopen/Materia/pull/1304) - [update] lti embed preview page updates
- [#1303](https://github.com/ucfopen/Materia/pull/1303) - adds ssl/tls for docker dev
- [#1302](https://github.com/ucfopen/Materia/pull/1302) - Bump lodash from 4.17.15 to 4.17.19
- [#1300](https://github.com/ucfopen/Materia/pull/1300) - Implements dotenv to streamline config  & install
- [#1268](https://github.com/ucfopen/Materia/pull/1268) - View and restore previously saved qsets in creator
- [#1260](https://github.com/ucfopen/Materia/pull/1260) - #1216 Inline Help on My Widgets Page

Not obvious in the PR list

- Fix to prevent samesite cookie set to 'None' when *not* using HTTPS
- Changes how LTI's are configured to allow for a catch-all default config that shares settings with all LTI consumers while allowing specific individual configurations for specific tool consumers.
- new localhost/dev page to place development-only tools and utilities when FUEL_ENV is in 'development' mode
- install wizard used to set config vars is completely removed due to heavy simplification of config in .env 
- LTI: adds support for `ext_outcome_data_values_accepted=url` which, with Canvas, will allow students and faculty to follow a link directly from their score results in the gradebook back to the materia score page for that play 👍 

To be paired with https://github.com/ucfopen/Materia-Server-Client-Assets/pull/108
(MSCA alpha release is tagged in this repo's package.json, no need to `yarn link` it)

Release TODOs:

- [x] review materia code
- [x] review msca code
- [x] manually test qset history
- [x] manually test lti score passback 
- [x] manually test lti gradebook score lookup
- [x] manually lti embed preview (obo next i think?)
- [x] manually lti faculty view template of lti assignment in Canvas
- [x] manually test access level when canceling demotion
- [x] manually test url params returning from picker (I think this was for obo next, but canvas may be possible)
- [x] determine ansible deploy changes / requirements & evaluate if docker in production is a shorter path
- [x] merge and tag
- [x] package & deploy to QA
- [ ] build release notes (include gif or screen shots of new features)
- [ ] deploy to prod
- [x] convert travis to github actions

### CODE REVIEW

**Review by iturgeon — APPROVED** [review]
So far, looks good to me.  Added a todo list for release in PR description

**Comment by clpetersonucf** [comment]
@iturgeon not sure about this one: 
> manually lti faculty view template of lti assignment in Canvas
This is separate from the LTI embed preview, which also got updated - right?

**Comment by iturgeon** [comment]
@clpetersonucf lti in embed vs regular preview urls:  I've lost the context for this in my head.

looking at https://github.com/ucfopen/Materia/pull/1304 - I think previewing just never supported embed mode before - and   This is probably a good motivation to overhaul the lti test provider to make it easier to add to.

File: fuel/app/config/theme.php [hunk 1]
```diff
@@ -35,9 +35,7 @@
 	 * The theme search paths.  They are searched in the order given.  You can add paths
 	 * on the fly via Theme::add_path($path) or Theme::add_paths(array($path1, $path2));
 	 */
-	'paths' => array(
-		APPPATH.'themes',
-	),
+	'paths' => explode(',', APPPATH.'themes'.','.($_ENV['THEME_PACKAGE'] ? PKGPATH.$_ENV['THEME_PACKAGE'] : '')),
```
Comment (clpetersonucf) [inline fuel/app/config/theme.php:38]: We don't have `THEME_PACKAGE` currently defined in the env configs on this branch, is it required?

File: fuel/app/classes/controller/widgets.php [hunk 2]
```diff
@@ -290,7 +291,7 @@ protected function retired()
 		Js::push_group(['angular', 'materia']);
 	}
 
-	protected function no_attempts($inst)
+	protected function no_attempts(string $inst, bool $is_embedded)
```
Comment (FrenjaminBanklin) [inline fuel/app/classes/controller/widgets.php:294]:
```suggestion
	protected function no_attempts(object $inst, bool $is_embedded)
```
`$inst` is definitely always going to be an object here.

This function is only called from one place: Currently line 390 in this same file, where it is never provided with a second argument, meaning `$is_embedded` is always missing.

That's likely unintended, so line 390 will need to be modified to pass `$is_embedded` as well.

**Review by FrenjaminBanklin — CHANGES_REQUESTED** [review]
Some picked nits with documentation of the .env files - mostly spelling corrections, etc.

Tested on QA.
The qset history UI malfunctions if the page is too narrow, but that isn't a significant issue.

The 'no attempts remaining' page was broken at some point and needs fixing, but it looks like the solution is pretty simple.

Otherwise, looks okay.

File: fuel/app/classes/controller/widgets.php (hunk 2 above)
Comment (clpetersonucf) [reply fuel/app/classes/controller/widgets.php]: Resolved via a58442c9601d6e9692686cb6cbee9407fe819880

**Review by FrenjaminBanklin — APPROVED** [review]
The handful of things I found appear to be resolved.

Looks good to me.

**Pull request merged**